src/Benraz.Infrastructure.Files/GCP/GcpBucketFilesService.cs

```diff
@@ -0,0 +1,331 @@
+            }
+            catch (RpcException ex) when (ex.StatusCode == Grpc.Core.StatusCode.NotFound)
+            {
+            }
+        }
+    }
+
+    /// <summary>
+    /// Deletes a folder and all its contents.
+    /// </summary>
+    /// <param name="path">Path to the folder.</param>
+    /// <returns>Task.</returns>
+    public async Task DeletePathAsync(string path)
+    {
+        string folderPath = path.EndsWith("/") ? path : path + "/";
+        
+        await DeleteAllAsync(folderPath);
+
+        try
+        {
+            await _storageControl.DeleteFolderAsync(GetFoldersResource(folderPath));
+        }
+        catch (RpcException ex) when (ex.StatusCode == Grpc.Core.StatusCode.NotFound)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Returns file URI.
+    /// </summary>
+    /// <param name="fileName">File name.</param>
+    /// <returns>File URI.</returns>
+    public string GetUri(string fileName)
+    {
+        return $"https://storage.googleapis.com/{_settings.BucketName}/{Uri.EscapeDataString(fileName)}";
+    }
+
+    /// <summary>
+    /// Move file from source to destination.
+    /// </summary>
+    /// <param name="fileName">File name.</param>
+    /// <param name="sourcePath">Source path.</param>
+    /// <param name="destinationPath">Destination path.</param>
+    /// <returns>Transfer status.</returns>
+    public async Task<bool> MoveFileAsync(string fileName, string sourcePath, string destinationPath)
+    {
+        var sourceObjectName = $"{sourcePath}/{fileName}".Trim('/');
+        var destinationObjectName = $"{destinationPath}/{fileName}".Trim('/');
+        await _storageClient.CopyObjectAsync(_settings.BucketName, sourceObjectName, _settings.BucketName, destinationObjectName);
+        await _storageClient.DeleteObjectAsync(_settings.BucketName, sourceObjectName);
+
+        return true;
+    }
+
+    /// <summary>
+    /// Create new folder by path.
+    /// </summary>
+    /// <param name="folderName">Folder name.</param>
+    /// <param name="path">Path.</param>
+    /// <returns>Creation status.</returns>
+    public async Task<bool> CreateFolderAsync(string folderName, string path)
+    {
+        var folderObjectName = $"{path}/{folderName}/".Trim('/');
+
+        StorageControlClient storageControl = StorageControlClient.Create();
+
+        var request = new CreateFolderRequest
+        {
+            Parent = BucketName.FormatProjectBucket("_", _settings.BucketName),
+            FolderId = folderObjectName
+        };
+
+        await storageControl.CreateFolderAsync(request);
+
+        return true;
+    }
+
+    /// <summary>
+    /// Returns signed file URI.
+    /// </summary>
+    /// <param name="fileName">File name.</param>
+    /// <param name="expiresInHours">Expires in hours.</param>
+    /// <returns>Signed file URI.</returns>
+    public string GetSignedUri(string fileName, int expiresInHours = 1)
```

Comment:
@fuadagazade you are missing some of the task requiremnts
As you can see this is old implementation that return GET access signed url (for read)
the require was to generate a CREATE access signed url (for write) as well, because i want to be able to create empty object, then send the client the url and let the client use this url to upload the file directly to reduce traffic in our platform.

Please add Create endpoint or  (i think better) update the interface to do the following:
public string GetSignedUri(string fileName, int expiresInHours = 1, **HttpMethod httpMethod = HttpMethod.Get**)

   var url = urlSigner.Sign(
            bucket: _settings.BucketName,
            objectName: fileName,
            duration: TimeSpan.FromHours(expiresInHours),
            httpMethod: **httpMethod** 
        );

You may update the Blob?

Reply:
Done